### ISSUE TITLE
Kruspe - 2019 - One-Way Prototypical Networks

### ISSUE THREAD

**Issue by wagaken** [wagaken/PAPPER_SUMMARY#8]
arXiv:1906.00820v1

**Comment by wagaken** [comment]
### Introduction
**Problem setting**
when training prototypical few-shot learning for a single class,  no relative distance to other classes can be calculated via softmax.

**Approach**
introduce a “null class” centered around zero, and enforce centering with batch normalization  (because Cosine or Euclidean distances are not bounded, we cannot set a threshold for class belonging)

**Characteristics**
construct a prototypical network that only requires positive support examples (one-way prototypical network)

### Method

![image](https://user-images.githubusercontent.com/49052500/148677666-85adb0aa-b2f6-4e55-ae8c-369907d59209.png)

![image](https://user-images.githubusercontent.com/49052500/148677675-c7cdf476-6322-4322-8eac-3c16fb98bfe8.png)

Queries are judged on whether they are more probable to belong to the (positive) class we are interested in rather than the null class.

As the model includes batch normalization layers, we can assume that the centroid of the whole latent space will be close to
the zero point in all embedding dimensions. 

 A query’s distance to the centroid of the positive class is therefore compared to its norm (i.e. its distance to the zero centroid) to determine whether it is more likely to belong to the positive class or to the null class encompassing everything.

propose an extension to prototypical networks: Modeling classes by their distribution instead of their centroid only.
A query’s class assignment is then determined by its likelihood under each class’s distribution

Instead of assuming a centroid at zero for the null class, we now assume a multivariate normal distribution with a mean at zero and a standard deviation of 1 (this is not guaranteed exactly, though: Batch normalization enforces a centering at zero and a standard deviation of 1 but does not enforce Gaussian-ness). Visualization is provided in figure 3. 

![image](https://user-images.githubusercontent.com/49052500/148678057-268131fd-2924-4daa-a76a-366690b2e66a.png)

Of course, this idea has one obvious drawback: The model requires a sufficient number of support examples to be able to determine a salient standard deviation over them, particularly as the embedding dimensions are treated independently. We hypothesize that beyond a certain number of support examples, implementing class’s distributions instead of their centroids only will enable the model to distinguish between them better.
